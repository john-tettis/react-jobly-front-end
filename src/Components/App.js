import React, {useState, useEffect} from 'react';
import {BrowserRouter, Route,Switch} from 'react-router-dom'
import '../Styles/App.css'
import Home from './Home'
import Navbar from './Navbar'
import CompaniesList from './CompaniesList'
import LoginForm from './Forms/LoginForm'
import SignupForm from './Forms/SignupForm'
import UserContext from '.././UserContext'
import CompanyInfo from './CompanyInfo'
import JobList from './JobList'
import ProfileInfo from './ProfileInfo'

function App() {
  let [user, setUser]= useState(null)
  useEffect(()=>{
    let storedUser= JSON.parse(window.localStorage.getItem('user'))
    setUser(storedUser)
  },[])
  useEffect(()=>{
        window.localStorage.setItem('user', JSON.stringify(user))
  },[user])
 

  return (
    <BrowserRouter>
        <Switch>
            <UserContext.Provider value={{user, setUser}}>
                <Navbar/>
                <Route exact path='/'>
                    <Home/>
                </Route>
                <Route exact path='/signup'>
                    <SignupForm/>
                </Route>
                <Route exact path='/login'>
                    <LoginForm/>
                </Route>
                <Route exact path='/companies'>
                    <CompaniesList/>
                </Route>
                <Route exact path='/companies/:handle'>
                    <CompanyInfo/>
                </Route>
                <Route exact path='/jobs'>
                    <JobList/>
                </Route>
                <Route exact path='/profile'>
                    <ProfileInfo/>
                </Route>
            </UserContext.Provider>
        </Switch>
    </BrowserRouter>
  );
}

export default App;
