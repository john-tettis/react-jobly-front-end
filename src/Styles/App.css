html, #root {
  background-color: #f5f5f5;
}

#root {
  max-width: 100%;
}

* {
  margin: 0;
  padding: 0;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}

.navbar {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  background-color: #24031e;
  height: 60px;
}

.navbar a {
  margin: auto 1rem;
  text-decoration: none;
  color: white;
  font-weight: 800;
}

.navbar-container {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
}

.navbar-container a {
  font-weight: 500;
}

.home {
  width: 100%;
  height: calc(100vh - 60px);
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
}

.home-container {
  text-align: center;
  margin-bottom: 1rem;
}

.home-title {
  color: #9B4746;
  font-size: 4rem;
}

.home-text {
  font-size: 1.2em;
  vertical-align: middle;
}

.home-button-container {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
}

.home-greet {
  margin-top: 5rem;
  font-size: 2rem;
  font-weight: 600;
}

.button-link {
  text-decoration: none;
  color: white;
  background-color: #EFB554;
  border: none;
  border-radius: 4px;
  margin: 1rem;
}

.button-link div {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  padding: .5rem 1rem;
  height: 100%;
  width: 100%;
  border-radius: 4px;
}

.button-link.success {
  background: #298329;
}

.button-link.disabled {
  -webkit-filter: brightness(0.7);
          filter: brightness(0.7);
}

.button-link:hover {
  color: white;
}

.button-link:hover > div {
  background-color: rgba(0, 0, 0, 0.2);
}

.button-block {
  width: 100%;
  margin: 0;
  margin-bottom: 1rem;
}

.form {
  width: 30rem;
}

@media screen and (max-width: 768px) {
  .form {
    width: 90%;
  }
}

.form-title {
  margin-top: 2rem;
  text-align: center;
}

.form input {
  margin-top: 5px;
}

.mcard {
  padding: 1rem;
  margin: 1rem;
  min-height: 6rem;
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -ms-flex-pack: distribute;
      justify-content: space-around;
  border-radius: 4px;
  background-color: white;
  -webkit-box-shadow: 1px 1px 4px #2e202b;
          box-shadow: 1px 1px 4px #2e202b;
}

.mcard.clickable:hover {
  -webkit-box-shadow: 3px 3px 8px #2e202b;
          box-shadow: 3px 3px 8px #2e202b;
}

.mcard-title {
  font-size: 1.2rem;
}

.mcard-container {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

.mcard-description {
  color: #3d3a3a;
}

.card-list {
  margin: auto;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  width: 60rem;
}

@media screen and (max-width: 965px) {
  .card-list {
    width: 40rem;
  }
}

@media screen and (max-width: 768px) {
  .card-list {
    width: 90%;
  }
}

.company-info {
  margin: auto;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  width: 60rem;
}

@media screen and (max-width: 965px) {
  .company-info {
    width: 40rem;
  }
}

@media screen and (max-width: 768px) {
  .company-info {
    width: 90%;
  }
}

.company-info-title {
  margin-top: 2rem;
}

.company-info-description {
  text-align: center;
  margin: 1rem auto .5rem auto;
}

.profile-info {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
}

.profile-info-title {
  margin-top: 2rem;
}

.search-bar {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-pack: distribute;
      justify-content: space-around;
  width: 100%;
}

.search-bar-input {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  padding: .5rem;
  margin: 1rem;
  margin-left: 0;
  font-size: 1.3rem;
  border-radius: 4px;
}

.search-bar .button-link {
  margin-right: 0;
  width: 8rem;
}

.search-bar .button-link > div {
  font-size: 1.3rem;
}

.job-list {
  margin: auto;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  width: 60rem;
}

@media screen and (max-width: 965px) {
  .job-list {
    width: 40rem;
  }
}

@media screen and (max-width: 768px) {
  .job-list {
    width: 90%;
  }
}

.error-message {
  color: red;
}
/*# sourceMappingURL=App.css.map */