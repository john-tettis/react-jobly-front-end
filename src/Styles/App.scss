
$dark:rgb(36, 3, 30);
$primary:#9B4746;
$secondary:#EFB554;
$muted:rgb(46, 32, 43);

@mixin tablet{
    @media screen and (max-width:768px){
        @content
    }
};
@mixin card-point{
    @media screen and (max-width:965px){
        @content
    }
};

@mixin list{
    margin:auto;
    display:flex;
    flex-direction: column;
    align-items:center;
    width:60rem;
    @include card-point{
        width:40rem;
    }
    @include tablet{
        width:90%;
    }
};


html, #root{
    background-color:#f5f5f5;
}
#root{
    max-width:100%;
}

*{
    margin:0;
    padding:0;
    box-sizing: border-box;
}



.navbar{
    display:flex;
    justify-content:space-between;
    align-items:center;
    background-color:$dark;
    height:60px;
    a{
        margin: auto 1rem;
        text-decoration:none;
        color:white;
      
        font-weight:800;
    }

    &-container{
        display:flex;
        a{
            font-weight:500;
        }
    }
   
}


.home{
    width:100%;
    height: calc(100vh - 60px);
    display:flex;
    align-items:center;
    justify-content:center;

    &-container{
        text-align: center;
        margin-bottom:1rem;
    }
    &-title{
        color:$primary;
        font-size:4rem;
    }
    &-text{
        font-size: 1.2em;
        vertical-align:middle;
    }
    &-button-container{
        display:flex;
        justify-content:center;
    }
    &-greet{
        margin-top:5rem;
        font-size:2rem;
        font-weight:600;
    }

}



.button-link{
    text-decoration: none;
    color:white;
    background-color:$secondary;
    border:none;
    border-radius: 4px;
    margin:1rem;
    div{
        display:flex;
        align-items:center;
        justify-content:center;
        padding:.5rem 1rem;
        height:100%;
        width:100%;
        border-radius:4px;
    }
    &.success{
        background: rgb(41, 131, 41);
    }
    &.disabled{
        filter:brightness(.7)
    }
    &:hover{ 
        color:white;
        &>div{
        // backdrop-filter: brightness(.5);
        background-color:rgba(0,0,0,.2);
        }
    }
}

.button-block{
    width:100%;
    margin:0;
    margin-bottom:1rem;
}

.form{
    width:30rem;
    @include tablet{
        width:90%;
    }
    &-title{
        margin-top:2rem;
        text-align:center;
    }
    input{
        margin-top:5px;
    }
}

.mcard{
    padding:1rem;
    margin:1rem;
    min-height:6rem;
    width:100%; 
    display:flex;
    flex-direction:column;
    justify-content:space-around; 
    border-radius: 4px;
    background-color:white;
    box-shadow: 1px 1px  4px $muted;

    &.clickable{
        &:hover{
            box-shadow: 3px 3px 8px $muted;
        }

    }
    &-title{
        font-size:1.2rem;
    }
    &-container{
        display:flex;
        justify-content:space-between;
        align-items:center;
    }
    &-description{
        color:#3d3a3a;
    }

}

.card-list{
    @include list;
}

.company-info{
    @include list;
    &-title{
        margin-top:2rem;
    }
    &-description{
        text-align:center;
        margin:1rem auto .5rem auto;
    }
}

.profile-info{
    display:flex;
    align-items:center;
    flex-direction: column;

    &-title{
        margin-top:2rem;
    }
}


.search-bar{
    display:flex;
    justify-content:space-around;
    width:100%;
    &-input{
        display:flex;
        padding: .5rem;
        margin:1rem;
        margin-left:0;
        font-size:1.3rem;
        border-radius: 4px;
    }
    .button-link{
        margin-right: 0;
        width:8rem;
        &>div{
            font-size:1.3rem;
        }
    }
}

.job-list{
    @include list;
}

.error-message{
    color:red;
}
