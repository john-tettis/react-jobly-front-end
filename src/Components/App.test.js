import { render, screen } from '@testing-library/react';
import App from './App';

test('renders App', () => {
  render(<App />);

});

test('renders App to snapshot', () => {
  let {asFragment} = render(<App />);
  expect(asFragment()).toMatchSnapshot();

});
