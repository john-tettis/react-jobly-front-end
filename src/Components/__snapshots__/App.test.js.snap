// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App to snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      Jobly
    </a>
    <div
      class="navbar-container"
    >
      <a
        href="/login"
      >
        Login
      </a>
      <a
        href="/signup"
      >
        Signup
      </a>
    </div>
  </nav>
  <div
    class="home"
  >
    <div
      class="home-container"
    >
      <h1
        class="home-title"
      >
        Jobly
      </h1>
      <p
        class="home-text"
      >
        Find your perfect career match through our extensive listings!
      </p>
      <div
        class="home-button-container"
      >
        <a
          class="button-link"
          href="/login"
        >
          <div>
            Login
          </div>
        </a>
        <a
          class="button-link"
          href="/signup"
        >
          <div>
            Sign Up
          </div>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
